common/src/main/java/com/wynntils/commands/UpdateCommand.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.commands;
+
+import com.mojang.brigadier.builder.LiteralArgumentBuilder;
+import com.mojang.brigadier.context.CommandContext;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.commands.CommandBase;
+import com.wynntils.core.managers.UpdateManager;
+import java.util.concurrent.CompletableFuture;
+import net.minecraft.ChatFormatting;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import net.minecraft.network.chat.TextComponent;
+
+public class UpdateCommand extends CommandBase {
+    @Override
+    public LiteralArgumentBuilder<CommandSourceStack> getBaseCommandBuilder() {
+        return Commands.literal("update").executes(this::update);
+    }
+
+    private int update(CommandContext<CommandSourceStack> context) {
+        if (WynntilsMod.isDevelopmentEnvironment()) {
+            context.getSource()
+                    .sendFailure(new TextComponent("Development environment detected, cannot update!")
+                            .withStyle(ChatFormatting.DARK_RED));
+            WynntilsMod.error("Development environment detected, cannot update!");
+            return 0;
+        }
+
+        WynntilsMod.info("Attempting to fetch Wynntils update.");
+        CompletableFuture<UpdateManager.UpdateResult> completableFuture = UpdateManager.tryUpdate();
+
+        completableFuture.whenComplete((result, throwable) -> {
```

Comment:
If this takes a long time, does it hang the minecraft chat?

Reply:
I did not test, but I don't think so.

Reply:
If so, it might be better to just send "Starting download" as a return for the command, and then do a "send chat message" thing with "Download complete" by the downloading thread when it is done.

Reply:
We need to test this

Reply:
One way to test this would be to set the URL to a server name that does not resolve. Usually DNS takes a handful of seconds before a missing name results in an error.

Reply:
I could just thread sleep the request.

Reply:
@magicus  https://github.com/Wynntils/Artemis/commit/738774f71e8e79bbd6abc0eb58c31b985527d582 I only tested with thread sleep so this might not be actually required, but we can have it anyways I guess.